### DISCUSSION TITLE
react-pdf internals

### DISCUSSION THREAD

**Discussion by MoadMerroun** [diegomura/react-pdf#2001] [Q&A] [answered]
### The official docs don't talk about internals (architecture, design choices), is there any resource explaining that ?

**Comment by diegomura** [comment, marked as answer]
Official docs are for public apis. There isn't anything specific for project/code architecture. Is there something specific you would like to know?

**Reply by MoadMerroun** [reply]
I am doing a research about it and I have to understand how it works under the hood (just like understanding how JSX and reconciliation in react work under the hood, I need to know how this renderer work)

**Reply by diegomura** [reply]
There aren't any docs for it, nor here and not officially by React afaik. I recommend checking https://github.com/diegomura/react-pdf/blob/master/packages/renderer/src/renderer.js and it's package here. It isn't very big

**Reply by MoadMerroun** [reply]
OK. Thanks a lot 👍👍
